mass-deployer/internal/parser/validators.go

```diff
@@ -29,7 +33,51 @@ func validateNetwork(network string) error {
 	return nil
 }
 
-func validateVMs(vms []deployer.Vms, nodeGroups []deployer.NodesGroup, sskKeys map[string]string) error {
+func validateNodeGroups(nodeGroups []massDeployer.NodesGroup, mnemonic, network string) error {
+	tfPluginClient, err := deployer.NewTFPluginClient(mnemonic, peer.KeyTypeSr25519, network, "", "", "", 0, false, false)
+	if err != nil {
+		return err
+	}
+
+	var multiErr error
+	var wg sync.WaitGroup
+	var lock sync.Mutex
+
+	for _, group := range nodeGroups {
+		wg.Add(1)
+		go func(group massDeployer.NodesGroup) {
+			defer wg.Done()
+			nodeGroupName := strings.TrimSpace(group.Name)
```

Comment:
extract the logic for the validation of a nodeGroup into a local scoped function or even a function